buildartifacts-gradle-plugin/src/main/java/com/google/cloud/buildartifacts/gradle/plugin/BuildArtifactsGradlePlugin.java

```
@@ -0,0 +1,103 @@
+import org.gradle.api.internal.artifacts.repositories.DefaultMavenArtifactRepository;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.publish.PublishingExtension;
+
+public class BuildArtifactsGradlePlugin implements Plugin<Project> {
+
+	private CredentialProvider credentialProvider = new DefaultCredentialProvider();
+
+    @Override
+    public void apply(Project project) {
+        project.afterEvaluate(p -> {
+            project.getRepositories().all(this::configureBuildArtifactsRepositories);
+            final PublishingExtension publishingExtension = project.getExtensions().findByType(PublishingExtension.class);
+            if (publishingExtension != null) {
+                publishingExtension.getRepositories().all(this::configureBuildArtifactsRepositories);
+            }
+        });
+    }
+
+    public void configureBuildArtifactsRepositories(ArtifactRepository repo)
+    {
+        try {
+            GoogleCredentials credentials = (GoogleCredentials)credentialProvider.getCredential();
+            AccessToken accessToken = credentials.getAccessToken();
+            String token = accessToken.getTokenValue();
+            
+            if (!(repo instanceof DefaultMavenArtifactRepository)) {
+                return;
+            }
+            final DefaultMavenArtifactRepository cbaRepo = (DefaultMavenArtifactRepository) repo;
+            final URI u = cbaRepo.getUrl(); 
+            if (u != null && u.getScheme() != null && u.getScheme().equals("buildartifacts")) {
+                String httpURL = u.toString().replaceFirst("buildartifacts://", "https://");
+                cbaRepo.setUrl(URI.create(httpURL));
+                BuildArtifactsPasswordCredentials crd = new BuildArtifactsPasswordCredentials("oauth2accesstoken", token);
+                cbaRepo.setConfiguredCredentials((Credentials)crd);
+            }
+        } catch (IOException e) {
+            System.out.println(e.toString());
```

I think we should propagate this Exception so that the user will see an appropriate message when there's a problem. I think you could subclass one of the Exceptions from the Gradle API: https://docs.gradle.org/current/javadoc/org/gradle/api/package-summary.html

---

done